### DISCUSSION TITLE
monai.apps.load_from_mmar Returns ValueError

### DISCUSSION THREAD

**Discussion by BK0x7C8** [Project-MONAI/MONAI#6712] [Q&A]
Hello,

I'd like to download and use pre-trained models from clara using mmar. I follow the instructions from MONAI documentation, but I receive errors.

- Error 1, when using api=True flag:

```
unet_model = load_from_mmar(
    item="clara_pt_prostate_mri_segmentation_1",
    mmar_dir=r'C:\Users\BKeoh\Desktop',
    map_location=device,
    pretrained=True,
    api=True,
    version=1)
```

`ValueError: api query returns no item for pattern {'name': 'clara_pt_prostate_mri_segmentation_1'}.  Please change or shorten it.`

- Error 2, when using api=False flag:

```
unet_model = load_from_mmar(
    item="clara_pt_prostate_mri_segmentation_1",
    mmar_dir=r'C:\Users\BKeoh\Desktop',
    map_location=device,
    pretrained=True,
    api=False,
    version=1)
```

`2023-07-12 15:00:39,655 - ERROR - Download failed from https://api.ngc.nvidia.com/v2/models/nvidia/med/clara_pt_prostate_mri_segmentation/versions/1/zip to [C:\Users\BKeoh\AppData\Local\Temp\tmphwl083qa\clara_pt_prostate_mri_segmentation_1.zip.](file:///C:/Users/BKeoh/AppData/Local/Temp/tmphwl083qa/clara_pt_prostate_mri_segmentation_1.zip.)`

I am unsure, if there is anything necessary for me to have stored locally and passed to the api via mmar_dir, e.g., a config.json.

Thanks in advance.

**Comment by wyli** [comment]
thanks for the question, those are deprecated, please consider the latest model zoo bundles https://github.com/Project-MONAI/model-zoo/tree/dev/models, main use cases: https://github.com/Project-MONAI/tutorials/tree/main/model_zoo

**Reply by BK0x7C8** [reply]
Dear wyli,

thanks for your reply. I am using the MONAI Zoo model of the Spleen segmentation. However, I have a different use case, e.g., I want to segment a different structure. I am currently using the .pt file from the Zoo model and I am unsure, if I am using it correctly:

```
# set the flag, it will change the initial model object 
self.PRETRAINED = True

def model(self, pretrained):
    # Init model 
    model = UNet(
        spatial_dims=3,
        in_channels=1,
        out_channels=2,
        channels=(16, 32, 64, 128, 256), 
        strides=(2, 2, 2, 2),
        num_res_units=2,
        norm=Norm.BATCH,
    ).to(self.device)
    
    # Use pretrained weights
    if pretrained:          
        file = r'C:\Users\BKeoh\Desktop\spleen_ct_segmentation_v0.5.3\spleen_ct_segmentation\models\model.pt'
        model.load_state_dict(torch.load(file))
  
        # Pattern to match the parameter names
        pattern = ['model.0.*', 'model.1.submodule.0.*']
  
        # Iterate through model parameters
        for name, param in model.named_parameters():
            # Check if the parameter name matches the pattern using fnmatch
            for pat in pattern:
                if fnmatch.fnmatch(name, pat):
                    param.requires_grad = False
                    
                    print(f'Using pretrained model with {len(pattern)} frozen layers')
    
    else:            
        print('Using randomly init. model')
  
    return model
```

with using `pattern = ['model.0.*', 'model.1.submodule.0.*']`, I am freezing the weights of the first two layers of the Unet encoder. Can you give me a confirmation, if this how I can work with the model weights? I am not very experienced in doing transfer learning 'from scratch'. 

Many thanks!
Bjørn
